omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/MockTaskFactoryWrapper.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web;
+
+import org.openmrs.module.webservices.helper.TaskFactoryWrapper;
+import org.openmrs.scheduler.SchedulerException;
+import org.openmrs.scheduler.TaskDefinition;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MockTaskFactoryWrapper extends TaskFactoryWrapper {
+	
+	public List<String> registeredTasks = new ArrayList<String>();
+	public List<String> scheduledTasks = new ArrayList<String>();
+	
+	@Override
+	public void saveTaskDefinition(TaskDefinition task) {
+		registeredTasks.add(task.getName());
+	}
+	
+	@Override
+	public void unloadTask(TaskDefinition task) throws SchedulerException {
+		scheduledTasks.remove(task.getName());
+		registeredTasks.remove(task.getName());
+	}
+	
+	@Override
+	public void scheduleTask(TaskDefinition task) throws SchedulerException {
+		boolean contains = false;
+		for (String taskDef : scheduledTasks) {
+			if (taskDef.equals(task.getName())) {
+				contains = true;
+				break;
+			}
+		}
+		if (!contains) {
+			scheduledTasks.add(task.getName());
+		}
```

Comment:
Did you intentionally decide to silently ignore a call for scheduling a task which does not exist?

Reply:
Here, I am doing same as ModuleAction Test. I using mock collections to maintain registered and scheduled tasks. When REST call arrives at Resource, that will call these MockTaskFactoryWrapper class methods. Here I am mocking that tasks and assigning to the collections to check the status from Test Class.